### PR TITLE
qa/tests: Add additional required ceph-ansible vars due to upstream changes

### PR DESCRIPTION
**Pull request by vasukulkarni**
@yuriw  want to test with this? Just those 2 tests?

Signed-off-by: Vasu Kulkarni <vasu@redhat.com>

### CODE REVIEW

File: qa/suites/rgw/hadoop-s3a/s3a-hadoop-v28.yaml [hunk 1]
```diff
@@ -18,6 +18,10 @@ overrides:
         journal_collocation: true
         osd_auto_discovery: false
         journal_size: 1024
+        ceph_stable_release: luminous
+        osd_scenario: collocated
+        ceph_origin: repository
+        ceph_repository: community
```
Comment (andrewschoen) [inline qa/suites/rgw/hadoop-s3a/s3a-hadoop-v28.yaml]: ``ceph_repository`` should be set to ``dev`` if you're installing packages from shaman and ``ceph_dev`` is no longer valid

File: qa/suites/rgw/hadoop-s3a/s3a-hadoop.yaml [hunk 2]
```diff
@@ -9,15 +9,18 @@ overrides:
...
         ceph_dev: true
         ceph_dev_key: https://download.ceph.com/keys/autobuild.asc
         ceph_origin: upstream
-        journal_collocation: true
         osd_auto_discovery: false
         journal_size: 1024
+        ceph_stable_release: luminous
+        osd_scenario: collocated
+        ceph_origin: repository
+        ceph_repository: community
```
Comment (andrewschoen) [inline qa/suites/rgw/hadoop-s3a/s3a-hadoop.yaml]: same here, this should be ``dev`` and ``ceph_dev`` removed

**Comment by vasukulkarni** [comment]
@yuriw test with https://github.com/ceph/teuthology/pull/1110

File: qa/suites/rgw/hadoop-s3a/s3a-hadoop-v28.yaml (hunk 1 above)
Comment (vasukulkarni) [reply qa/suites/rgw/hadoop-s3a/s3a-hadoop-v28.yaml]: @andrewschoen are you sure? This one I copied from other ceph-ansible suite which is working, I will check the logs there. ceph_dev is internal set in task.

Comment (andrewschoen) [reply qa/suites/rgw/hadoop-s3a/s3a-hadoop-v28.yaml]: @vasukulkarni I'm pretty sure that if ``ceph_repository: community`` is set you'll be getting packages from download.ceph.com. I'm not surprised the job is passing, but I bet you aren't installing what you think you are. Yeah, ``ceph_dev`` isn't an option at all anymore either.

Comment (vasukulkarni) [reply qa/suites/rgw/hadoop-s3a/s3a-hadoop-v28.yaml]: thanks @andrewschoen , It is from download.ceph.com as you pointed out http://qa-proxy.ceph.com/teuthology/vasu-2017-09-12_20:21:30-ceph-ansible-master-distro-basic-ovh/1624063/teuthology.log and I will remove the ceph_dev key.

**Comment by vasukulkarni** [comment]
@andrewschoen mind looking at it again and approving it ?  I have updated the other suite as well, In the task ceph_dev_branch is set to sha, hopefully that is correct.

File: qa/suites/ceph-ansible/smoke/basic/2-ceph/ceph_ansible.yaml [hunk 3]
```diff
@@ -8,7 +8,6 @@ overrides:
           global:
             osd default pool size: 2
             mon pg warn min per osd: 2
-        ceph_dev: true
         ceph_dev_key: https://download.ceph.com/keys/autobuild.asc
```
Comment (andrewschoen) [inline qa/suites/ceph-ansible/smoke/basic/2-ceph/ceph_ansible.yaml]: ``ceph_dev_key`` isn't needed either

File: qa/suites/rgw/hadoop-s3a/s3a-hadoop-v28.yaml [hunk 4]
```diff
@@ -9,15 +9,18 @@ overrides:
             osd default pool size: 2
             osd pool default pg num: 128
             osd pool default pgp num: 128
-            debug rgw: 20 
+            debug rgw: 20
             debug ms: 1
         ceph_test: true
-        ceph_dev: true
         ceph_dev_key: https://download.ceph.com/keys/autobuild.asc
         ceph_origin: upstream
```
Comment (andrewschoen) [inline qa/suites/rgw/hadoop-s3a/s3a-hadoop-v28.yaml]: You should remove any instance of ``ceph_origin: upstream`` as it's not a valid option for ``ceph_origin`` and you've already got it defined as ``repository``.

Comment (vasukulkarni) [reply qa/suites/rgw/hadoop-s3a/s3a-hadoop-v28.yaml]: @andrewschoen thanks will remove both of them (dev_key and _origin).

**Comment by yuriw** [comment]
@vasukulkarni @smithfarm @theanalyst needs backporting luminous and maybe jewel?

**Comment by theanalyst** [comment]
there is a merge conflict which needs resolving

**Comment by vasukulkarni** [comment]
a) tracker: http://tracker.ceph.com/issues/21497 for rgw issue
b) the tests are working as expected for smoke: http://pulpito.ceph.com/vasu-2017-09-21_22:03:45-ceph-ansible-master-distro-basic-ovh/

**Pull request merged**